### PR TITLE
Add start type to DMON

### PR DESCRIPTION
**Pull request by huber-nicolas**

### CODE REVIEW

File: file-formats/dmon/dmon-v1.json [hunk 1]
```diff
@@ -111,10 +111,31 @@
         "System Daemon"
       ],
       "enumDescriptions": [
-        "Application daemon",
-        "System daemon"
+        "Application Daemon",
+        "System Daemon"
```
Comment (schneidermic0) [inline file-formats/dmon/dmon-v1.json:115]: Is titlecase intended for "Daemon" in descriptions?

Comment (huber-nicolas) [reply file-formats/dmon/dmon-v1.json:115]: Daemon is an official SAP Term, I think that was the reason why I wrote it in upper case

File: file-formats/dmon/type/zif_aff_dmon_v1.intf.abap [hunk 2]
```diff
@@ -37,44 +37,70 @@ INTERFACE zif_aff_dmon_v1
...
+
+  CONSTANTS:
+    "! <p class="shorttext">start Type</p>
+    "! start type
+    BEGIN OF co_start_type,
+      "! <p class="shorttext">Automatically started Daemon</p>
+      "! Automatically started Daemon
+      automatically_started       TYPE ty_start_type VALUE 'A',
+      "! <p class="shorttext">Explicitly started Daemon (same user)</p>
+      "! Explicitly started Daemon (Same User)
```
Comment (schneidermic0) [inline file-formats/dmon/type/zif_aff_dmon_v1.intf.abap:61]:
```suggestion
      "! <p class="shorttext">Explicitly Started Daemon (same user)</p>
      "! Explicitly started Daemon (same user)
```

File: file-formats/dmon/type/zif_aff_dmon_v1.intf.abap [hunk 3]
```diff
@@ -37,44 +37,70 @@ INTERFACE zif_aff_dmon_v1
...
+      explicitly_started_sameuser TYPE ty_start_type VALUE 'H',
+      "! <p class="shorttext">Explicitly started Daemon (different user, deprecated)</p>
+      "! Explicitly started Daemon (different user, deprecated)
```
Comment (schneidermic0) [inline file-formats/dmon/type/zif_aff_dmon_v1.intf.abap:64]:
```suggestion
      "! <p class="shorttext">Explicitly Started Daemon (different user, deprecated)</p>
      "! Explicitly started Daemon (different user, deprecated)
```

File: file-formats/dmon/type/zif_aff_dmon_v1.intf.abap [hunk 4]
```diff
@@ -37,44 +37,70 @@ INTERFACE zif_aff_dmon_v1
...
+   "! $values {@link zif_aff_dmon_v1.data:co_start_type}
+   "! $default {@link zif_aff_dmon_v1.data:co_start_type.automatically_started}
+    ty_start_type TYPE c LENGTH 1.
+
+  CONSTANTS:
+    "! <p class="shorttext">start Type</p>
+    "! start type
+    BEGIN OF co_start_type,
+      "! <p class="shorttext">Automatically started Daemon</p>
+      "! Automatically started Daemon
```
Comment (schneidermic0) [inline file-formats/dmon/type/zif_aff_dmon_v1.intf.abap:58]:
```suggestion
      "! <p class="shorttext">Automatically Started Daemon</p>
      "! Automatically started Daemon
```

**Comment by huber-nicolas** [comment]
Why are the markdown links check red?

**Review by schneidermic0 — APPROVED** [review]
LGTM

File: file-formats/dmon/dmon-v1.json [hunk 5]
```diff
@@ -111,10 +111,31 @@
...
+      ],
+      "enumTitles": [
+        "Automatically Started Daemon",
+        "Explicitly Started Daemon (same user)",
+        "Explicitly Started Daemon (different user, deprecated)"
+      ],
+      "enumDescriptions": [
+        "Automatically started Daemon",
+        "Explicitly started Daemon (same User)",
+        "Explicitly started Daemon (different user, deprecated)"
```
Comment (albertmink) [inline file-formats/dmon/dmon-v1.json:136]:
```suggestion
      "enumTitles": [
        "Automatically Started Daemon",
        "Explicitly Started Daemon (same user)",
        "Explicitly Started Daemon (different user, deprecated)"
      ],
      "enumDescriptions": [
        "Automatically started Daemon",
        "Explicitly started Daemon (same user)",
        "Explicitly started Daemon (different user, deprecated)"
```

Comment (albertmink) [reply file-formats/dmon/dmon-v1.json:136]: I guess this is a typo, right?

Comment (huber-nicolas) [reply file-formats/dmon/dmon-v1.json:136]: yes, thanks

**Pull request merged**